### ISSUE TITLE
"borg diff" doesn't show empty files as difference. (Unexpected behavior)

### ISSUE THREAD

**Issue by tykonin** [borgbackup/borg#3688]
If I have 2 archives differing by empty files (but also having another difference) the "**borg diff**" wont show the empty files as difference. If the only difference are some empty files, I see no output at all.  

### Example / how to reproduce:  
Let "/home/user/bk" be Borg repository with no encryption (didn't tried with encryption).  
Create archive of the directory "/home/user/foo" with 
`borg create /home/user/bk::arc0 /home/user/foo`
Now add some empty files in "~/foo", 
`touch /home/user/foo/file1`  
Create another archive of the dir " ~/foo "  

 `borg create /home/user/bk::arc1 /home/user/foo `   

Comparing the two archives (arc0 and arh1) with:  
`borg diff /home/user/bk::arc0 arc1`  
It shows nothing.  
___
So far this way I reproduced it on CentOS 7 with Borg 1.1.4 from EPEL and on Manjaro with Borg 1.1.4 from the official repo.

The expected behaviour according to my interpretation of the documentation (Usage->borg diff->examples) is that it should show that we added one file (namely "file1"), no matter if it's the only difference or there are another differences (doesn't work in both cases).

**Comment by ThomasWaldmann** [comment]
master branch:
```
$ borg init -e none repo
$ mkdir foo
$ borg create repo::a foo
$ touch foo/empty
$ borg create repo::b foo
$ borg diff repo::a b
added           0 B foo/empty
```

**Comment by ThomasWaldmann** [comment]
1.1.4:
```
$ borg init -e none repo
$ rm -rf foo
$ mkdir foo
$ borg create repo::a foo
$ touch foo/empty
$ borg create repo::b foo
$ borg diff repo::a b
```

**Comment by ThomasWaldmann** [comment]
The fixed master was a side effect of the refactor in #2842.

**Comment by ThomasWaldmann** [comment]
fixed by #3692.
